nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -839,7 +841,9 @@ def training_step(self, dataloader_iter):
             ignore_virtual=True
         ):
             if (
-                self.cfg.get('defer_embedding_wgrad_compute', False) and self.mcore_gpt
+                self.cfg.get('defer_embedding_wgrad_compute', False)
+                and self.mcore_gpt
+                and not self.use_mcore_dist_optim
```

Comment:
Why does MCore distributed optimizer + defer wgrad support not work?

Reply:
It works. The functionality done in NeMo for Apex DistOpt is already done in MCore when using Mcore optim. So when using MCore Opt with NeMo, we are doing the same operation twice, so removing it from NeMo when using MCore Opt.